### ISSUE TITLE
Pydantic hook does not include optional dotenv dependency

### ISSUE THREAD

**Issue by johnthagen** [pyinstaller/pyinstaller-hooks-contrib#523]
**Describe the bug**
* Which hook/library isn't working? Pydantic
* Does the error get raised while building or when running?

```
ModuleNotFoundError: No module named 'dotenv'

ImportError: python-dotenv is not installed, run `pip install pydantic[dotenv]`
[3019] Failed to execute script 'pcb_dr' due to unhandled exception!
```

**To Reproduce**

A minimal example file:
```py
from pydantic import BaseSettings


class MySettings(BaseSettings):
    FOO: str

    class Config:
        env_file = '.env'


def main() -> None:
    settings = MySettings()
    print(settings.FOO)


if __name__ == "__main__":
    main()
```

And create a small `.env` file:

```
FOO=BAR
```

PyInstaller command:
```
pyinstaller --onefile main.py
```

**Expected behavior**

Running outside Pyinstaller:

```
$ python main.py
BAR
```

**Desktop (please complete the following information):**
 - OS: macOS
 - Python Version: 3.10.8
 - Version of `pyinstaller-hooks-contrib`: 2022.13
 - Version of PyInstaller: 5.6.2

**Additional context**

```
$ pip list
Package                   Version
------------------------- -------
altgraph                  0.17.3
macholib                  1.16.2
pip                       22.3.1
pydantic                  1.10.2
pyinstaller               5.6.2
pyinstaller-hooks-contrib 2022.13
python-dotenv             0.21.0
setuptools                65.6.3
typing_extensions         4.4.0
wheel                     0.38.4
```

You can see here in the Pydantic source that the import is hidden:

- https://github.com/pydantic/pydantic/blob/v1.10.2/pydantic/env_settings.py#L326

Documentation on this feature:

- https://pydantic-docs.helpmanual.io/usage/settings/#dotenv-env-support

**Comment by rokm** [comment]
The underlying problem is probably that you are using the default, cython-compiled `pydantic` package (https://pydantic-docs.helpmanual.io/install/#performance-vs-package-size-trade-off), so the dependencies in extensions that are not explicitly listed in the hook's hidden imports are not picked up automatically.

Feel free to extend the hook's hidden imports with `dotenv`, and open a PR. 

The question is, should the hook collect optional dependencies? I think it doesn't hurt, because it would be collected with non-cythonized `pydantic`, provided it was available in the first place. So users who don't need it and work with build environment that has only required dependencies installed won't be affected by its addition to hidden imports.

**Comment by johnthagen** [comment]
> The question is, should the hook collect optional dependencies?

This was my question as well. I wasn't sure what the precedent was for the pyinstaller hooks.

I did confirm that, as expected, this also solved this issue:

```
pyinstaller --onefile  --hidden-import=dotenv main.py
```

**Comment by johnthagen** [comment]
I can confirm that this issue is fixed in the `2022.14` hooks release.
